common/include/common/logger/trivial_logger.hpp

```diff
@@ -0,0 +1,78 @@
+#ifndef RDMN_COMMON_TRIVIAL_LOGGER_HPP
+#define RDMN_COMMON_TRIVIAL_LOGGER_HPP
+
+#include <sstream>
+#include <iostream>
+
+#include <rdmn-engine/public_api.hpp>
+#include <rdmn-engine/logger/LoggerType.hpp>
+
+namespace rdmn::core::log {
+
+/**
+ * @brief 
+ * Current trivial_logger implementation is created to mimic Boost::Log semantics
+ * in order to reduce th needed changes when doing refactoring of many,
+ * and I mean MANY lines of logging code.
+ * It is 
+ * - not efficient
+ * - not robust
+ * - not practical
+ * - and overall a necessary evil
+ * 
+ * This implementation will be replacedby a proper logger, which will be supplied to classes which require it,
+ * when propper logging refactoring is done.
+ * 
+ */
+
+R_PUBLIC_API void log_trace(std::string &&str);
+R_PUBLIC_API void log_info(std::string &&str);
+R_PUBLIC_API void log_debug(std::string &&str);
+R_PUBLIC_API void log_warn(std::string &&str);
+R_PUBLIC_API void log_error(std::string &&str);
+R_PUBLIC_API void log_fatal(std::string &&str);
+
+struct trivial_logger {
+    R_PUBLIC_API trivial_logger(LoggerType type);
+    R_PUBLIC_API ~trivial_logger();
+
+    R_PUBLIC_API trivial_logger(const trivial_logger &other) = delete;
+    R_PUBLIC_API trivial_logger &operator=(const trivial_logger &other) = delete;
+
+    R_PUBLIC_API trivial_logger(trivial_logger &&other) = delete;
+    R_PUBLIC_API trivial_logger &operator=(trivial_logger &&other) = delete;
+
+    template <typename Val>
+    R_PUBLIC_API trivial_logger &&operator<<(Val &&v) && {
+        iss_ << v;
+        return std::move(*this);
+    }
+
+    template <size_t Size>
+    R_PUBLIC_API trivial_logger &&operator<<(const char (&cstr)[Size]) && {
+        for (size_t i = 0; i < Size - 1; ++i) {
+            iss_ << cstr[i];
+        }
+
+        if (cstr[Size - 1] != '\0')
+            iss_ << '\0';
```

Comment:
This function looks pretty strange, why in the loop above it's `i < Size - 1` and then you have zero terminator check and addtition?